querqy-for-lucene/querqy-solr/src/main/java/querqy/solr/IndexRewriterContainer.java

```diff
@@ -0,0 +1,493 @@
+        }
+    }
+
+    protected Map<String, Object> readConfigurationFromDocument(final SolrDocument doc) {
+        final var fieldValue = doc.getFieldValue(FIELD_DATA);
+        if (fieldValue == null) {
+            throw new SolrException(INVALID_STATE, String.format("Unexpected null value encountered in field '%s'", FIELD_DATA));
+        } else if (fieldValue instanceof IndexableField) {
+            var byteRef = ((IndexableField) fieldValue).binaryValue();
+            return JsonUtil.readMapFromJson(byteRef.utf8ToString());
+        } else {
+            throw new SolrException(INVALID_STATE, String.format("Unexpected field type '%s' encountered in field '%s'", fieldValue.getClass().getName(), FIELD_DATA));
+        }
+    }
+
+    protected void writeConfigurationToDocument(final Map<String, Object> configuration, final SolrInputDocument document) {
+        final var byteArrayOutputStream = new ByteArrayOutputStream();
+        JsonUtil.writeJson(configuration, byteArrayOutputStream);
+        document.addField(FIELD_DATA, byteArrayOutputStream.toByteArray());
+    }
+
+    protected Optional<SolrDocument> configurationDocumentForRewriter(final SolrIndexSearcher searcher, final String coreName, final String rewriterId) throws IOException {
+        try (final LocalSolrQueryRequest req = new LocalSolrQueryRequest(searcher.getCore(), new ModifiableSolrParams())) {
+
+            final QueryResult result = new QueryResult();
+
+            final QueryCommand queryCommand = new QueryCommand();
+            queryCommand.setQuery(new TermQuery(new Term(FIELD_DOC_ID, configurationDocumentId(coreName, rewriterId))));
+            queryCommand.setLen(1);
+
+            searcher.search(result, queryCommand);
+
+            final SolrDocumentList docs = toSolrDocumentList(result.getDocList(),
+                    searcher,
+                    searcher.getSchema(),
+                    new SolrReturnFields(
+                            new String[]{
+                                    FIELD_DOC_ID,
+                                    FIELD_REWRITER_ID,
+                                    FIELD_CORE_NAME,
+                                    FIELD_CONF_VERSION,
+                                    FIELD_DATA
+                            },
+                            req
+                    )
+            );
+
+            return docs.stream().findFirst();
+        }
+    }
+
+    protected List<SolrDocument> allConfigurationDocuments(final SolrIndexSearcher searcher, final String coreName) throws IOException {
+        final ArrayList<SolrDocument> configurationDocuments = new ArrayList<>();
+        try (final LocalSolrQueryRequest req = new LocalSolrQueryRequest(searcher.getCore(), new ModifiableSolrParams())) {
+            final ResponseBuilder rsp = new ResponseBuilder(req,
```

Comment:
The `rsp` does not seem to be used, can we remove it?

Reply:
removed with 04027d14622d6b1e311ae0ad2c7a1b85aa647a51